src/realm/query_expression.hpp

```diff
@@ -2183,6 +2245,75 @@ class LinkCount : public Subexpr2<Int> {
     LinkMap m_link_map;
 };
 
+template <class oper, class TLeft>
+class SizeOperator : public Subexpr2<Int> {
+public:
+    SizeOperator(std::unique_ptr<TLeft> left)
+        : m_left(std::move(left))
+    {
+    }
+
+    SizeOperator(const SizeOperator& other, QueryNodeHandoverPatches* patches)
```

Comment:
This constructor type is usually public (`SimpleQuerySupport`, `columns<T>`). Anyway I can see that I can get away with having it private in this case if I rewrite clone() function.